zipkin/src/main/java/com/linecorp/armeria/internal/tracing/SpanContextUtil.java

```diff
@@ -16,32 +16,38 @@
 
 package com.linecorp.armeria.internal.tracing;
 
-import com.linecorp.armeria.common.RequestContext;
+import java.util.Collections;
+
 import com.linecorp.armeria.common.logging.RequestLog;
+import com.linecorp.armeria.common.tracing.RequestContextCurrentTraceContext;
 
 import brave.Span;
-import brave.Tracer;
-import brave.Tracer.SpanInScope;
-import io.netty.util.concurrent.FastThreadLocal;
+import brave.Tracing;
+import brave.propagation.CurrentTraceContext.Scope;
+import brave.propagation.TraceContext;
 
-/**
- * Utility for pushing and popping a {@link Span} via a {@link RequestContext}.
- */
 public final class SpanContextUtil {
 
     /**
-     * Sets up the {@link RequestContext} to push and pop the {@link Span} whenever it is entered/exited.
+     * Trace context scopes are often wrapped, for example to decorate MDC. This creates a dummy scope to
+     * ensure {@link RequestContextCurrentTraceContext#INSTANCE} is configured, even if it is wrapped.
      */
-    public static void setupContext(FastThreadLocal<SpanInScope> threadLocalSpan, RequestContext ctx, Span span,
-                                    Tracer tracer) {
-        ctx.onEnter(unused -> threadLocalSpan.set(tracer.withSpanInScope(span)));
-        ctx.onExit(unused -> {
-            final SpanInScope spanInScope = threadLocalSpan.get();
-            if (spanInScope != null) {
-                spanInScope.close();
-                threadLocalSpan.remove();
-            }
-        });
+    public static void ensureScopeUsesRequestContext(Tracing tracing) {
```

Comment:
Perhaps we could move this method into `RequestContextCurrentTraceContext`? It seems more relevant to `RequestContextCurrentTraceContext`. Then we could also move `PingPongExtra` and its test class to `common.tracing` and make `PingPongExtra` package-local.

Reply:
very nice